src/views/ServiceRequest.js

```diff
@@ -98,30 +100,31 @@ class ServiceRequest extends Component {
     const value = target.type === 'checkbox' ? target.checked : target.value
     const name = target.name
 
-    this.setState({
-      [name]: value
-    })
+    const form = Object.assign({}, this.state.form)
+    form[name] = value
+
+    this.setState({ form })
```

Comment:
This way of setting the state is necessary because we are using a nested object.
```js
// this does not work
this.setState({form: {[name]: value}})
```
We are first using [Object.assign()](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) to get a shallow copy of the current form object.
We then mutate that copy with the value we need to change.
Finally we replace the whole form object that is currently in state with the mutated copy.